components/match_ticker/commons/match_ticker_display_components_new.lua

```diff
@@ -0,0 +1,366 @@
+local Countdown = require('Module:Countdown')
+local DateExt = require('Module:Date/Ext')
+local LeagueIcon = require('Module:LeagueIcon')
+local Logic = require('Module:Logic')
+local Lua = require('Module:Lua')
+local Table = require('Module:Table')
+local Timezone = require('Module:Timezone')
+local StreamLinks = require('Module:Links/Stream')
+
+local HighlightConditions = Lua.import('Module:HighlightConditions')
+
+local OpponentLibraries = require('Module:OpponentLibraries')
+local Opponent = OpponentLibraries.Opponent
+local OpponentDisplay = OpponentLibraries.OpponentDisplay
+
+local VS = 'VS'
+local SCORE_STATUS = 'S'
+local CURRENT_PAGE = mw.title.getCurrentTitle().text
+local HIGHLIGHT_CLASS = 'tournament-highlighted-bg'
+local TOURNAMENT_DEFAULT_ICON = 'Generic_Tournament_icon.png'
+local NOW = os.date('%Y-%m-%d %H:%M', os.time(os.date('!*t') --[[@as osdateparam]]))
+
+---Display class for the header of a match ticker
+---@class NewMatchTickerHeader
+---@operator call(string|number|nil): NewMatchTickerHeader
+---@field root Html
+local Header = Class.new(
+	function(self, text)
+		self.root = mw.html.create('div')
+			:addClass('infobox-header')
+			:wikitext(text)
+	end
+)
+
+---@param class string?
+---@return NewMatchTickerHeader
+function Header:addClass(class)
+	self.root:addClass(class)
+	return self
+end
+
+---@return Html
+function Header:create()
+	return mw.html.create('div'):node(self.root)
+end
+
+---Display class for matches shown within a match ticker
+---@class NewMatchTickerVersus
+---@operator call(table): NewMatchTickerVersus
+---@field root Html
+---@field match table
+local Versus = Class.new(
+	function(self, match)
+		self.root = mw.html.create('div'):addClass('versus')
+		self.match = match
+	end
+)
+
+---@return Html
+function Versus:create()
+	local bestof = self:bestof()
+	local scores = self:scores()
+	local upperText, lowerText
+
+	if bestof then
+		upperText = scores or VS
+		lowerText = bestof
+	elseif scores then
+		upperText = scores
+		lowerText = VS
+	end
+
+	if not lowerText then
+		return self.root:wikitext(VS)
+	end
+
+	return self.root
+		:node(mw.html.create('div')
+			:addClass('versus-upper')
+			:node(upperText or VS)
+		):node(mw.html.create('div')
+			:addClass('versus-lower')
+			:wikitext('(' .. lowerText .. ')')
+		)
+end
+
+---@return string?
+function Versus:bestof()
+	local bestof = tonumber(self.match.bestof) or 0
+	if bestof > 0 then
+		return Abbreviation.make('Bo' .. bestof, 'Best of ' .. bestof)
+	end
+end
+
+---@return string?
+---@return string?
+function Versus:scores()
+	if self.match.date > NOW then
+		return
+	end
+
+	local scores = {}
+
+	Array.forEach(self.match.match2opponents, function(opponent, opponentIndex)
+		local score = Logic.isNotEmpty(opponent.status) and opponent.status ~= SCORE_STATUS and opponent.status
+			or tonumber(opponent.score) or -1
+
+		table.insert(scores, score)
+	end)
+
+	return table.concat(scores, ' : ')
+end
```

Comment:
is there a reason we are dropping score2 support (for reset matches)

Reply:
I wasn't sure if it was being used in Dota2, but if it is I'll bring it back

Reply:
they have at least 4 pages with bracket reset match:
https://liquipedia.net/dota2/index.php?search=insource%3ARxMBR
(potetntially many more old ones with the old bracket format)

Reply:
Reused the existing Versus from MatchTicker/DisplayComponents instead to not have duplicates